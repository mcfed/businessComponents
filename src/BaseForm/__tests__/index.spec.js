import React from 'react';
import {render, shallow, mount} from 'enzyme';

import BaseForm, {AdvancedForm} from '../index';

describe('advancedForm autosubmit', () => {
  it('base test', () => {
    const wrapper = shallow(
      <AdvancedForm autoSubmitForm={true} onSearch={jest.fn()}></AdvancedForm>
    );
    const instance = wrapper.instance();

    instance.getForm().setFieldsValue({a: 1, b: 2});
    expect(wrapper.prop('onSearch')).toHaveBeenCalled();
  });
  it('autosubmit form false ', () => {
    const wrapper = shallow(
      <AdvancedForm autoSubmitForm={false} onSearch={jest.fn()}></AdvancedForm>
    );
    const instance = wrapper.instance();

    instance.getForm().setFieldsValue({a: 1, b: 2});
    expect(wrapper.prop('onSearch')).not.toHaveBeenCalled();
  });
});

describe('baseform test', () => {
  it('render', () => {
    const wrapper = render(
      <BaseForm
        autoSubmitForm={true}
        onSearch={jest.fn}
        hideRequiredMark={true}>
        <div>111</div>
        <div>222</div>
      </BaseForm>
    );
    expect(wrapper).toMatchSnapshot();
  });
  const wrapper = shallow(
    <BaseForm autoSubmitForm={true} onSearch={jest.fn} hideRequiredMark={true}>
      <div>111</div>
      <div>222</div>
    </BaseForm>
  );
  it('base test', () => {
    expect(wrapper.prop('hideRequiredMark')).toEqual(true);
  });
});
