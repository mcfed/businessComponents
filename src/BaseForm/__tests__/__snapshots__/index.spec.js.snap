// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`baseform test render 1`] = `
<form
  class="ant-form ant-form-horizontal ant-form-hide-required-mark"
>
  <div>
    111
  </div>
  <div>
    222
  </div>
</form>
`;
